### ISSUE TITLE
Heightfield modification & PVD

### ISSUE THREAD

**Issue by zemlifr** [NVIDIAGameWorks/PhysX#621]
Hello, I am using heightfield modification based on snippet provided here: https://gameworksdocs.nvidia.com/PhysX/4.1/documentation/physxguide/Manual/Geometry.html#heightfield-modification

It seem to work fine, however I don't see changes propagated to PVD which makes debugging a bit hard for me.

I can see there is some heightfield specific code in 

`void PvdMetaDataBinding::releaseAndRecreateGeometry(PvdDataStream& inStream, const PxShape& inObj, PxPhysics& /*ownerPhysics*/, PsPvd* pvd)`

which is called, but it doesn't seem to work. 
Is there something that could be fixed in physx code, or there is issue on PVD side?

**Comment by pdherasciec** [comment]
Hello, heightfields get exported correctly both using PVD and OVD, would you mind sharing the code for the PVD setup? The samples are not automatically outputting to PVD nor OVD.

Here are some screens of using Kit and PVD2, to view heightfields from a heightfield UT
![heightfield_ut](https://user-images.githubusercontent.com/115007036/220104243-8b902bea-f042-49f0-9305-f7aec782d141.JPG)
![heightfield_ut_pvd](https://user-images.githubusercontent.com/115007036/220104284-7e5058be-466b-4747-bf5e-10e6459798ca.JPG)

**Comment by pdherasciec** [comment]
Realizing you want to actually modify the heightfield. Need to check up on this. I think the OVD viewer in Kit does not allow, or rather does not take such modifications into account. Need to see if to make a ticket for this.

**Comment by pdherasciec** [comment]
Just FYI this is being worked on next. Thanks for exposing the issue!

**Comment by zemlifr** [comment]
So bug is on the side of PVD and there is nothing I can do to patch it myself?

**Comment by MunoDevelop** [comment]
@pdherasciec hi , sorry to ask here. does pvd visualizer support real time debug render or you read after from ovd file? i tried before but nothing i can see in realtime

**Comment by pdherasciec** [comment]
> @pdherasciec hi , sorry to ask here. does pvd visualizer support real time debug render or you read after from ovd file? i tried before but nothing i can see in realtime

Correct, at the moment the OVD reader in Kit/Composer that is omni.physx.pvd (or the OmniPVD extension) allows the import of an already finished OVD file.
